community/5_mins_rag_no_gpu/main.py

```diff
@@ -22,56 +22,44 @@
 
 import streamlit as st
 import os
+from langchain_nvidia_ai_endpoints import ChatNVIDIA, NVIDIAEmbeddings
+from langchain.text_splitter import CharacterTextSplitter
+from langchain_community.document_loaders import DirectoryLoader
+from langchain_community.vectorstores import FAISS
+import pickle
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.prompts import ChatPromptTemplate
 
-st.set_page_config(layout = "wide")
+st.set_page_config(layout="wide")
 
+# Component #1 - Document Upload
 with st.sidebar:
     DOCS_DIR = os.path.abspath("./uploaded_docs")
     if not os.path.exists(DOCS_DIR):
         os.makedirs(DOCS_DIR)
     st.subheader("Add to the Knowledge Base")
     with st.form("my-form", clear_on_submit=True):
-        uploaded_files = st.file_uploader("Upload a file to the Knowledge Base:", accept_multiple_files = True)
+        uploaded_files = st.file_uploader("Upload a file to the Knowledge Base:", accept_multiple_files=True)
         submitted = st.form_submit_button("Upload!")
 
     if uploaded_files and submitted:
         for uploaded_file in uploaded_files:
             st.success(f"File {uploaded_file.name} uploaded successfully!")
-            with open(os.path.join(DOCS_DIR, uploaded_file.name),"wb") as f:
+            with open(os.path.join(DOCS_DIR, uploaded_file.name), "wb") as f:
                 f.write(uploaded_file.read())
 
-############################################
 # Component #2 - Embedding Model and LLM
-############################################
+llm = ChatNVIDIA(model="meta/llama3-70b-instruct")
+document_embedder = NVIDIAEmbeddings(model="NV-Embed-QA", model_type="passage")
```

Comment:
NV-Embed-QA -> nvidia/nv-embedqa-e5-v5 (that's the new default)

Reply:
Ack, fixed this and the requirements.txt